File: clase3/Cat.py
from Mouse import Mouse

class Cat:
    #Los atributos no son necesario de describir
    #Se describen en el constructor __init__()
    def __init__(self, id, name, sex, age, weight, color, texture):
        self.id = id
        self.name = name
        self.sex = sex
        self.age = age
        self.weight = weight
        self.color = color
        self.texture = texture
    
    def eat(self):
        print(self.name + " esta comiendo...")
    
    def move(self, direction):
        print(self.name + " se mueve a " + direction)

    def meow(self):
        print(self.name + "Dice Meow!")

    def purr(self):
        print(self.name + "Dice Purr!")

    def hunt_mice(self, hunted):
        print(self.name + " Cazó a " + hunted.name)
        return hunted

File: clase3/Cat.java

public class Cat{
    //Atributos
    int id;
    String name;
    String sex;
    double age;
    double weight;
    String color;
    String texture;

    //Métodos
    //Constructor
    public Cat(){ }
    public Cat(int id, String name,String sex, double age, double weight, String color, String texture){
        this.id = id;
        this.name = name;
        this.sex = sex;
        this.age = age;
        this.weight = weight;
        this.color = color;
        this.texture = texture;
    }

    //comunes - personalizados
    public void eat(){
        System.out.println(this.name + " esta comiendo...");
    }

    public void move(String direction){
        System.out.println(this.name + " se mueve a " + direction);
    }

    public void meow(){
        System.out.println(this.name + " Dice Meow...");
    }

    public void purr(){
        System.out.println(this.name + " Ronronea...");
    }

    public Mouse huntMice(Mouse hunted){
        System.out.println(this.name + " Cazó a " + hunted.name);
        return hunted;
    }

}